src/cppgc.rs

```diff
@@ -0,0 +1,294 @@
+// Copyright 2019-2021 the Deno authors. All rights reserved. MIT license
+
+use crate::platform::Platform;
+use crate::support::int;
+use crate::support::Opaque;
+use crate::support::SharedRef;
+use crate::support::UniqueRef;
+
+extern "C" {
+  fn cppgc__initialize_process(platform: *mut Platform);
+  fn cppgc__shutdown_process();
+
+  fn cppgc__heap__create(
+    platform: *mut Platform,
+    wrappable_type_index: int,
+    wrappable_instance_index: int,
+    embedder_id_for_garbage_collected: u16,
+  ) -> *mut Heap;
+  fn cppgc__heap__DELETE(heap: *mut Heap);
+  fn cppgc__make_garbage_collectable(
+    heap: *mut Heap,
+    obj: *mut (),
```

Comment:
**question:**  Would this work and/or make sense with a type parameter `T` and `obj: *mut T` as well as return value (presumably) `*mut T`? Same for `TraceFn<T>` and `DestroyFn<T>`.

Reply:
foreign types cannot have type parameters.
https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=75dc0a79442c4eaa9947b8f5c5fdbbf6